File: basic_queries/groq_llm_query.py
"""
python groq_llm_query.py "Hello! Who are you?"

Author: tdiprima
Version: 1.0
License: MIT
"""

__author__ = "tdiprima"
__version__ = "1.0"
__license__ = "MIT"

import argparse
import os

from groq import Groq


def main():
    # Set up command-line argument parser
    parser = argparse.ArgumentParser(
        description="Send a prompt to the Groq API and get a response."
    )
    parser.add_argument("prompt", type=str, help="The prompt to send to the Groq API.")
    args = parser.parse_args()

    # Initialize Groq client
    client = Groq(
        api_key=os.environ.get("GROQ_API_KEY"),
    )

    # Send chat completion request
    chat_completion = client.chat.completions.create(
        messages=[
            {
                "role": "user",
                "content": args.prompt,
            }
        ],
        model="compound-beta-mini",
        stream=False,
    )

    # Print the response
    print("\nResponse:")
    print(chat_completion.choices[0].message.content)


if __name__ == "__main__":
    main()

File: basic_queries/groq_llm_query.js
/*
node groq_llm_query.js "Explain the importance of fast language models"
*/

import Groq from "groq-sdk";

// Initialize Groq client
const groq = new Groq({ apiKey: process.env.GROQ_API_KEY });

// Main function
async function main() {
  // Retrieve the prompt from the command-line arguments
  const args = process.argv.slice(2);
  if (args.length === 0) {
    console.error("Please provide a prompt as a command-line argument.");
    process.exit(1);
  }
  const userPrompt = args.join(" ");

  try {
    // Send prompt to the Groq API
    const chatCompletion = await groq.chat.completions.create({
      messages: [
        {
          role: "user",
          content: userPrompt,
        },
      ],
      model: "compound-beta-mini",
    });

    // Print the API response
    console.log("Response:");
    console.log(chatCompletion.choices[0]?.message?.content || "No response received.");
  } catch (error) {
    console.error("Error:", error.message || error);
  }
}

// Run the main function
main();
